Chargebee/Classes/Purchase/CBPurchaseManager.swift

```diff
@@ -106,7 +106,32 @@ public extension CBPurchase {
     func purchaseProduct(product: CBProduct, customerId : String? = "" ,completion handler: @escaping ((_ result: Result<(status:Bool, subscriptionId:String?, planId:String?), Error>) -> Void)) {
         buyProductHandler = handler
         activeProduct = product.product
-        customerID = customerId ?? ""
+        self.customerInfo = CBCustomer(customerID: customerId ?? "", first_name: "", last_name: "", email: "")
+        guard CBAuthenticationManager.isSDKKeyPresent() else {
+            handler(.failure(CBPurchaseError.cannotMakePayments))
+            return
+        }
+
+        if !CBPurchase.shared.canMakePayments() {
+            handler(.failure(CBPurchaseError.cannotMakePayments))
+        } else {
+            authenticationManager.isSDKKeyValid { status in
+                if status {
+                    let payment = SKPayment(product: product.product)
+                    SKPaymentQueue.default().add(payment)
+
+                } else {
+                    handler(.failure(CBPurchaseError.invalidSDKKey))
+                }
+            }
+        }
+    }
+    func purchaseProduct(product: CBProduct, userInfo : CBCustomer? = nil, completion handler: @escaping ((_ result: Result<(status:Bool, subscriptionId:String?, planId:String?), Error>) -> Void)) {
+        buyProductHandler = handler
+        activeProduct = product.product
+        if let customerData = userInfo {
+            customerInfo = customerData
+        }
```

Comment:
Can we avoid duplicating the logic between the new method and existing one?

Reply:
Done refactored

Reply:
Do we need `if let..` here?

Reply:
updated